tests/test_replacer.py

```diff
@@ -520,44 +521,75 @@ def _issue_count(total=False):
         assert _issue_count(total=True) == [{"count": 1, "group_id": 1}]
 
     def test_query_time_flags(self) -> None:
-        project_ids = [1, 2]
+        project_ids = [1, 2, 3]
 
         assert errors_replacer.get_projects_query_flags(
             project_ids, ReplacerState.ERRORS
-        ) == (False, [],)
+        ) == (False, [], set())
 
         errors_replacer.set_project_needs_final(100, ReplacerState.ERRORS)
         assert errors_replacer.get_projects_query_flags(
             project_ids, ReplacerState.ERRORS
-        ) == (False, [],)
+        ) == (False, [], set())
 
         errors_replacer.set_project_needs_final(1, ReplacerState.ERRORS)
         assert errors_replacer.get_projects_query_flags(
             project_ids, ReplacerState.ERRORS
-        ) == (True, [],)
+        ) == (True, [], set())
         assert errors_replacer.get_projects_query_flags(
             project_ids, ReplacerState.EVENTS
-        ) == (False, [],)
+        ) == (False, [], set())
 
         errors_replacer.set_project_needs_final(2, ReplacerState.ERRORS)
         assert errors_replacer.get_projects_query_flags(
             project_ids, ReplacerState.ERRORS
-        ) == (True, [],)
+        ) == (True, [], set())
 
-        errors_replacer.set_project_exclude_groups(1, [1, 2], ReplacerState.ERRORS)
-        errors_replacer.set_project_exclude_groups(2, [3, 4], ReplacerState.ERRORS)
+        # Replacement types don't impact test, just checking if same ones show up
+        errors_replacer.set_project_exclude_groups(
+            1, [1, 2], ReplacerState.ERRORS, ReplacementType.EXCLUDE_GROUPS
+        )
+        errors_replacer.set_project_exclude_groups(
+            2, [3, 4], ReplacerState.ERRORS, ReplacementType.START_MERGE
+        )
+        assert errors_replacer.get_projects_query_flags(
+            project_ids, ReplacerState.ERRORS
+        ) == (
+            True,
+            [1, 2, 3, 4],
+            {ReplacementType.EXCLUDE_GROUPS, ReplacementType.START_MERGE},
+        )
+
+        errors_replacer.set_project_exclude_groups(
+            1, [1, 2], ReplacerState.ERRORS, ReplacementType.EXCLUDE_GROUPS
+        )
+        errors_replacer.set_project_exclude_groups(
+            2, [3, 4], ReplacerState.ERRORS, ReplacementType.EXCLUDE_GROUPS
+        )
+        errors_replacer.set_project_exclude_groups(
+            3, [5, 6], ReplacerState.ERRORS, ReplacementType.START_UNMERGE
+        )
         assert errors_replacer.get_projects_query_flags(
             project_ids, ReplacerState.ERRORS
-        ) == (True, [1, 2, 3, 4],)
+        ) == (
+            True,
+            [1, 2, 3, 4, 5, 6],
+            {
+                ReplacementType.EXCLUDE_GROUPS,
+                # start_merge should show up from previous setter on project id 2
+                ReplacementType.START_MERGE,
+                ReplacementType.START_UNMERGE,
+            },
+        )
```

Comment:
Could you also try to call `get_projects_query_flags` on a subset of project ids to ensure it returns only the relevant types for that project ?